packages/core/src/external/aws/lambda-logic/bundle-to-html.ts

```diff
@@ -1271,7 +1271,12 @@ function createObservationSocialHistorySection(observations: Observation[]) {
   const removeDuplicate = uniqWith(observationsSortedByDate, (a, b) => {
     const aDate = dayjs(a.effectiveDateTime).format(ISO_DATE);
     const bDate = dayjs(b.effectiveDateTime).format(ISO_DATE);
-    return aDate === bDate && a.code?.text === b.code?.text;
+    const aText = a.code?.text;
+    const bText = b.code?.text;
+    if (aText === undefined || bText === undefined) {
+      return false;
+    }
+    return aDate === bDate && aText === bText;
```

Comment:
we were missing section because when two section had the same date but both had undefined for code.text, this was evaluating to true, even though they were different sec tion

Reply:
this bug is actually repeated like dozen times across this file